test/GPUActivityAgentTest.cpp

```diff
@@ -181,89 +184,13 @@ TEST_F(GPUActivityAgentTest, validate_policy)
     EXPECT_NO_THROW(m_agent->validate_policy(policy));
     // validate policy defaults are applied
     ASSERT_EQ(m_num_policy, policy.size());
-    EXPECT_EQ(M_FREQ_MAX, policy[FREQ_MAX]);
-    EXPECT_EQ((policy[FREQ_MAX] + M_FREQ_MIN) / 2, policy[FREQ_EFFICIENT]);
-    EXPECT_EQ(0.5, policy[PHI]);
-
-    // all-NAN policy is accepted
-    // setup & load NAN policy
-    std::set<std::string> sig_set = {"LEVELZERO::GPU_CORE_FREQUENCY_EFFICIENT"};
-    EXPECT_CALL(*m_platform_io, signal_names()).WillOnce(Return(sig_set));
-    double fe_sig_val = M_FREQ_MAX - M_FREQ_MIN + 0.123;
-    EXPECT_CALL(*m_platform_io, read_signal("LEVELZERO::GPU_CORE_FREQUENCY_EFFICIENT",
-                                            GEOPM_DOMAIN_BOARD, 0)).WillOnce(Return(fe_sig_val));
-    policy = empty;
-    EXPECT_NO_THROW(m_agent->validate_policy(policy));
-    // validate policy defaults are applied
-    ASSERT_EQ(m_num_policy, policy.size());
-    EXPECT_EQ(M_FREQ_MAX, policy[FREQ_MAX]);
-    EXPECT_EQ(fe_sig_val, policy[FREQ_EFFICIENT]);
     EXPECT_EQ(0.5, policy[PHI]);
 
     // non-default policy is accepted
     // setup & load policy
     EXPECT_CALL(*m_platform_io, signal_names()).WillRepeatedly(Return(empty_set));
-    policy[FREQ_MAX] = M_FREQ_MAX;
-    policy[FREQ_EFFICIENT] = M_FREQ_MAX / 2;
     policy[PHI] = 0.1;
     EXPECT_NO_THROW(m_agent->validate_policy(policy));
-
-    // validate policy is modified as expected
```

Comment:
I know it no longer makes sense to validate the policy is modified as expect, however, it seems some error conditions can still be tested and validated, for example: I see the code still throws when Fe and Phi are out of range.

Reply:
Great catch, there was some overzealous cleanup that removed these existing test cases: https://github.com/geopm/geopm/blob/d1f2860844b39442b859911728aad887fe9345fa/test/GPUActivityAgentTest.cpp#L254

Reply:
To be honest, I didn't pay attention to this for the equivalent CPU-CA agent changes. Could you take a look at those tests as well? If any changes are needed there, that would be for a separate PR, of course.